### PR TITLE
Fetch list followers/followings

### PR DESCRIPTION
**Pull request by saraanej**
In this PR, I implemented the backend for the fetching of followers' and followings' lists. When visiting a profile, it is now possible to consult the list of followers and followings of a user, fetched from the database.

### CODE REVIEW

File: app/src/main/java/com/github/orkest/ui/FollowListActivity.kt
Comment (yasmineben510) [inline app/src/main/java/com/github/orkest/ui/FollowListActivity.kt:1]: Great modularization! Don't forget the java doc for the public functions

File: app/src/main/java/com/github/orkest/ui/FollowListViewModel.kt [hunk 1]
```diff
@@ -1,29 +1,40 @@
...
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import com.github.orkest.data.User
 import com.github.orkest.domain.FireStoreDatabaseAPI
+import java.util.concurrent.CompletableFuture
 
-class FollowListViewModel(val username: String, val isFollowersList: Boolean) {
+open class FollowListViewModel(val username: String, val isFollowersList: Boolean) {
 
     private val dbAPI = FireStoreDatabaseAPI()
```
Comment (yasmineben510) [inline app/src/main/java/com/github/orkest/ui/FollowListViewModel.kt:11]: the FirestoreDataBaseAPI is now static so you don't need to recreate an instance for it each time you can directly use its methods with FirestoreDataBaseAPI.fetch..

**Review by yasmineben510 — APPROVED** [review]
Great work overall, thank you! Only 2 remarks persist but other than that it is really great thank you!

File: app/src/androidTest/java/com/github/orkest/ViewModel/FollowListViewModelTest.kt [hunk 2]
```diff
@@ -34,32 +38,56 @@ class FollowListViewModelTest {
...
+            assertNotNull(followList)
+            assertEquals(2, followList.size)
+            assertTrue(followList[0].username == "user2")
+            assertTrue(followList[1].username == "user3")
         }
+        //Observes the list to finish fetching all the users from database and then apply the tests
+        followListLiveData.observeForever(followListObserver)
 
+        // Wait for LiveData to be updated
+        Thread.sleep(1000)
```
[the remark this answers is not in the document]
Comment (saraanej) [reply app/src/androidTest/java/com/github/orkest/ViewModel/FollowListViewModelTest.kt:68]: It's actually in milliseconds so the thread is sleeping for 1 second

File: app/src/main/java/com/github/orkest/ui/FollowListViewModel.kt [hunk 3]
```diff
@@ -1,29 +1,40 @@
...
-        val userList = mutableListOf<User>()
+    open fun retrieveFollowList(): LiveData<MutableList<User>> {
+        val userListLiveData = MutableLiveData<MutableList<User>>()
+
+        dbAPI.fetchFollowList(username, isFollowersList).thenApplyAsync { followList ->
+            val userList = mutableListOf<User>()
+            val userSearchFutures = mutableListOf<CompletableFuture<User>>()
 
-        dbAPI.fetchFollowList(username, isFollowersList).thenApply { followList ->
             followList.forEach { username ->
```
[the remark this answers is not in the document]
Comment (saraanej) [reply app/src/main/java/com/github/orkest/ui/FollowListViewModel.kt:23]: Will do so !

File: app/src/androidTest/java/com/github/orkest/ViewModel/FollowListViewModelTest.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (yasmineben510) [reply app/src/androidTest/java/com/github/orkest/ViewModel/FollowListViewModelTest.kt:68]: Ah my bad I read it ad 10 000 not 1000 sorry

**Pull request merged**